src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +129,32 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
+                            String addition = "_" + dockerBuildImage.getName();
+                            String safeEnvironmentVariable = Util.safeEnvironmentVariableName(environmentVariable + addition);
+                            System.out.println("SAFE: " + safeEnvironmentVariable);
+                            createSetComposeEnvironmentAction(safeEnvironmentVariable).execute(dockerBuildImage);
+                        }else{
+                            createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        }
                     }
                 });
             });
         });
 
         // Register shortened environment variables for `buildDockerImage` tasks created with the docker plugin
         pluginClasspathChecker.withPlugin(project, DockerPlugin.class, DockerPlugin.PLUGIN_ID, plugin -> {
```

Comment:
@vierbergenlars It does not go in here when an environmentVariable is given with the code.... Why is that? Is that something advanced I'm not aware of or am I overseeing something stupid?

Reply:
Yes, this block of code is only executed when the `DockerPlugin` is applied to `project`, like in this test: https://github.com/xenit-eu/alfresco-docker-gradle-plugin/blob/61038ee7b4c84c38ce774e19ec393c6bea54f0c2/src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java#L202

We can't try to do anything special for a task with name `buildDockerImage` without knowing that it exists (when the task does not exist, an exception would be thrown), so we limit it to the case where we know for sure that `buildDockerImage` exists, after the `DockerPlugin` has been applied.